lib/query/index.js

```diff
@@ -291,6 +291,11 @@ Query.prototype.parseValue = function parseValue(field, modifier, val) {
   "use strict";
   var self = this;
 
+  // Omit adding regex value to these modifiers
+  var omitRegExModifiers = ['$ne', 'greaterThan', '>', 'gt', 'greaterThanOrEqual',
+                            '>=', 'gte', '$gt', '$gte'
+                           ];
```

Comment:
@dmarcelino I didn't add the lessthan operator because waterline adapter test didn't complain. However, with my curiosity I just ran a few query for the lessthan operator with regex and it barf with error same as the above operators "BadValue Can't have RegEx as arg to predicate over field". I will add the lessthan to the list. Thanks.